### PR TITLE
RD-5228 Cypress 10 upgrade

### PR DESCRIPTION
**Pull request by kubama**
## Description
Cypress 10 upgrade

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/job/Stage-UI-System-Test/3866/ - :heavy_check_mark:  **pass**, the problematic test is failing on `master` too, corresponding Jira ticket: https://cloudifysource.atlassian.net/browse/RD-6388

https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/3901/ - :heavy_check_mark:  **pass**

https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/3917/ - :heavy_check_mark:  **pass**, the problematic test is failing on `master` too, corresponding Jira ticket: https://cloudifysource.atlassian.net/browse/RD-6455

 

## Documentation
N/A

### CODE REVIEW

File: test/jest/setupFiles.ts [hunk 1]
```diff
@@ -2,18 +2,8 @@ import { TextDecoder, TextEncoder } from 'util';
...
-            TextDecoder: typeof TextDecoder;
-        }
-    }
-}
-
 // NOTE: Workaround for https://github.com/jsdom/jsdom/issues/2524
 if (typeof TextEncoder !== 'undefined' && typeof TextDecoder !== 'undefined') {
     global.TextEncoder = TextEncoder;
-    global.TextDecoder = TextDecoder;
+    (global as any).TextDecoder = TextDecoder;
```
Comment (kubama) [inline test/jest/setupFiles.ts:8]: For some reason this stopped compiling, probably due to some indirect dependency upgrade. This simple change fixes it.

File: package.json [hunk 2]
```diff
@@ -29,20 +29,20 @@
         "check-types": "npm run check-types:frontend && npm run check-types:backend",
         "check-types:frontend": "tsc --build",
         "check-types:backend": "npm run --prefix backend check-types",
-        "cy:open": "cypress open -C node_modules/cloudify-ui-common-cypress/cypress.json",
```
Comment (qooban) [inline package.json:32]: Good that we can remove configuration file references from here 👍🏻

File: package.json [hunk 3]
```diff
@@ -145,8 +145,6 @@
     "// optionalDependencies": "NOTE: Development dependencies NOT required to build the package",
     "optionalDependencies": {
         "@cypress/code-coverage": "^3.9.12",
-        "@cypress/react": "^5.10.0",
-        "@cypress/webpack-dev-server": "^1.8.1",
```
Comment (qooban) [inline package.json:149]: So right now both packages are bundled in `cypress` package, right?

**Review by qooban — APPROVED** [review]
Looks cool!

File: package.json (hunk 3 above)
Comment (kubama) [reply package.json:149]: Yes

File: package.json [hunk 4]
```diff
@@ -29,19 +29,19 @@
         "check-types": "npm run check-types:frontend && npm run check-types:backend",
         "check-types:frontend": "tsc --build",
         "check-types:backend": "npm run --prefix backend check-types",
-        "cy:open": "cypress open -C node_modules/cloudify-ui-common-cypress/cypress.json",
-        "cy:open-ct": "cypress open-ct -C node_modules/cloudify-ui-common-cypress/cypress.json",
+        "cy:open": "cypress open --e2e --browser electron",
+        "cy:open-ct": "cypress open --component --browser electron",
```
Comment (qooban) [inline package.json:33]: Isn't `electron` a default choice?

File: test/cypress/integration/widgets/blueprints_spec.ts [hunk 5]
```diff
@@ -343,8 +340,13 @@ describe('Blueprints widget', () => {
 
                 const blueprintName = `${blueprintNamePrefix}_default_file`;
                 cy.get('input[name=blueprintName]').clear().type(blueprintName);
+                cy.interceptSp('PUT', `/blueprints/${blueprintName}`).as('createBlueprintRequest');
                 cy.get('.button.ok').click();
 
+                cy.wait('@createBlueprintRequest').its('response.body').should('contain', {
+                    main_file_name: 'read-secret-blueprint.yaml'
+                });
```
Comment (qooban) [inline test/cypress/integration/widgets/blueprints_spec.ts:348]: I like that change 👍🏻 

Using `blueprintSources` widget for `blueprints` widget verification was... Far from perfect 🙂

File: test/cypress/integration/widgets/filters_spec.ts [hunk 6]
```diff
@@ -115,6 +115,7 @@ describe('Filters widget', () => {
     function searchFilter(name: string) {
         cy.getSearchInput().clear().type(name);
         cy.get('.loading').should('not.exist');
+        cy.contains('table', name).should('be.visible');
```
Comment (qooban) [inline test/cypress/integration/widgets/filters_spec.ts:118]: Good idea 👍🏻

**Review by qooban — APPROVED** [review]
Looks good, added some minor comments, no blockers.

File: package.json (hunk 4 above)
Comment (kubama) [reply package.json:33]: It is for headless mode, for `cypress open` we would be getting a browser selection dialog without this setting. I found it inconvenient and unnecessary - the default for `cypress open` should be same as for headless, so I added this setting. Browser can be changed anytime inside the runner.

Comment (qooban) [reply package.json:33]: 👍🏻

**Pull request merged**